<launch>
    <!-- Common Interface -->
    <arg name="rate" default="10"/>

    <!-- Camera Settings -->
    <arg name="left" default="25000804"/>
    <arg name="left_camera_name" default="mv_$(arg left)"/>
    <arg name="left_calib_url"
    	default="file://${ROS_HOME}/camera_info/$(arg left_camera_name).yaml"/>

    <arg name="middle" default="25000851"/>
    <arg name="middle_camera_name" default="mv_$(arg middle)"/>
    <arg name="middle_calib_url"
    	default="file://${ROS_HOME}/camera_info/$(arg middle_camera_name).yaml"/>

    <arg name="right" default="25001110"/>
    <arg name="right_camera_name" default="mv_$(arg right)"/>
    <arg name="right_calib_url"
    	default="file://${ROS_HOME}/camera_info/$(arg right_camera_name).yaml"/>

    <arg name="camera" default="trinocular"/>
    <arg name="frame_id" default="$(arg camera)"/>
    <arg name="fps" default="$(arg rate)"/>
    <arg name="color" default="false"/>
    <arg name="aec" default="0"/>
    <arg name="cbm" default="false"/>
    <arg name="ctm" default="1"/>
    <arg name="dcfm" default="0"/>
    <arg name="hdr" default="false"/>
    <arg name="wbp" default="-1"/>
    <arg name="expose_us" default="5000"/>
    <arg name="gain_db" default="0.0"/>
    <arg name="mm" default="0"/>

    <!-- Node Settings -->
    <arg name="output" default="screen"/>

    <!-- Node -->
    <node pkg="bluefox2" type="bluefox2_multi_node" name="$(arg camera)" output="$(arg output)">
    	  <param name="num_cameras" type="int" value="3"/>
        <param name="camera0/camera_name" type="string" value="$(arg left_camera_name)"/>
        <param name="camera0/identifier" type="string" value="$(arg left)"/>
        <param name="camera0/calib_url" type="string" value="$(arg left_calib_url)"/>

        <param name="camera1/camera_name" type="string" value="$(arg middle_camera_name)"/>
        <param name="camera1/identifier" type="string" value="$(arg middle)"/>
        <param name="camera1/calib_url" type="string" value="$(arg middle_calib_url)"/>

        <param name="camera2/camera_name" type="string" value="$(arg right_camera_name)"/>
        <param name="camera2/identifier" type="string" value="$(arg right)"/>
        <param name="camera2/calib_url" type="string" value="$(arg right_calib_url)"/>

        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="fps" type="double" value="$(arg fps)"/>
        <param name="color" type="bool" value="$(arg color)"/>
        <param name="aec" type="int" value="$(arg aec)"/>
        <param name="cbm" type="bool" value="$(arg cbm)"/>
        <param name="ctm" type="int" value="$(arg ctm)"/>
        <param name="dcfm" type="int" value="$(arg dcfm)"/>
        <param name="hdr" type="bool" value="$(arg hdr)"/>
        <param name="wbp" type="int" value="$(arg wbp)"/>
        <param name="expose_us" type="int" value="$(arg expose_us)"/>
        <param name="gain_db" type="double" value="$(arg gain_db)"/>

        <remap from="~camera0/image_raw" to="~left/image_raw"/>
        <remap from="~camera1/image_raw" to="~middle/image_raw"/>
        <remap from="~camera2/image_raw" to="~right/image_raw"/>

        <remap from="~camera0/camera_info" to="~left/camera_info"/>
        <remap from="~camera1/camera_info" to="~middle/camera_info"/>
        <remap from="~camera2/camera_info" to="~right/camera_info"/>
    </node>
</launch>
